helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -326,32 +328,17 @@ public void close() {
     return newIdealStates;
   }
 
-  // Coordinate baseline recalculation and partial rebalance according to the cluster changes.
+  // Coordinate global rebalance and partial rebalance according to the cluster changes.
   protected Map<String, ResourceAssignment> computeBestPossibleAssignment(
       ResourceControllerDataProvider clusterData, Map<String, Resource> resourceMap,
-      Set<String> activeNodes, final CurrentStateOutput currentStateOutput)
+      Set<String> activeNodes, final CurrentStateOutput currentStateOutput,
+      RebalanceAlgorithm algorithm)
       throws HelixRebalanceException {
-    getChangeDetector().updateSnapshots(clusterData);
-    // Get all the changed items' information. Filter for the items that have content changed.
-    final Map<HelixConstants.ChangeType, Set<String>> clusterChanges =
-        getChangeDetector().getChangeTypes().stream()
-            .collect(Collectors.toMap(changeType -> changeType, changeType -> {
-              Set<String> itemKeys = new HashSet<>();
-              itemKeys.addAll(getChangeDetector().getAdditionsByType(changeType));
-              itemKeys.addAll(getChangeDetector().getChangesByType(changeType));
-              itemKeys.addAll(getChangeDetector().getRemovalsByType(changeType));
-              return itemKeys;
-            })).entrySet().stream().filter(changeEntry -> !changeEntry.getValue().isEmpty())
-            .collect(Collectors
-                .toMap(changeEntry -> changeEntry.getKey(), changeEntry -> changeEntry.getValue()));
-
-    // Perform Global Baseline Calculation
-    refreshBaseline(clusterData, clusterChanges, resourceMap, currentStateOutput);
-
-    // Perform partial rebalance
+    // Perform global rebalance for a new baseline assignment
+    globalRebalance(clusterData, resourceMap, currentStateOutput, algorithm);
```

Comment:
I think it will be better named 
```
final boolean waitForResult = _waitForGlobalRebalance;
triggerGlobalRebalance(parameters..., waitForResult);
```

Reply:
Is this for the next async PR?

Reply:
Yes, I'm reviewing all together with your potential next PR. I feel like a boolean variable explicitly here will make the code more readable (without looking into how `globalRebalance` does in details)

Reply:
Could you please hold on this review for the next PR. I cannot really do any change in this one : )